homeassistant/components/edl21/sensor.py

```diff
@@ -435,6 +434,12 @@ def __init__(self, electricity_id, obis, name, entity_description, telegram):
         }
         self._async_remove_dispatcher = None
         self.entity_description = entity_description
+        self._attr_name = name
+        self._attr_unique_id = f"{electricity_id}_{obis}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._electricity_id)},
+            name=self._electricity_id,
```

Comment:
Wat is `electricity_id`?
As it seems to be used for the unique ID as well, so... it isn't a name? Or?

Reply:
It is not a name, it is a unique hex value. The name of the device also needs to be unique, doesn't it? There is no other "unique property" delivered by the smart meter. I tried to set a human readable default name for the device at first, but ended up having all the entities of my two smart meters in one device. But maybe I did something else wrong...

Reply:
Identifiers need to be unique and static.
Name doesn't need to be unique, and can be adjusted by the user in the UI.
If there is no name available from the `message_body` then maybe it is better to set a default "pretty" name rather than a hex.

If you look at https://developers.home-assistant.io/blog/2022/07/10/entity_naming/ that should make it clearer.

Reply:
Ah, ok, understood...

I just reverted my local changes back to the state with the default name, I had it like this:

```
  identifiers={(DOMAIN, DEFAULT_DEVICE_NAME)},
  name=DEFAULT_DEVICE_NAME,
```
but it needs to be like this:

```
  identifiers={(DOMAIN, self._electricity_id)},
  name=DEFAULT_DEVICE_NAME,
```

Reply:
> It is not a name, it is a unique hex value.

In that case, I don't think this PR is right and should be reverted. A hex name is not a name of a device we should represent to a user, that is far from user-friendly.

Reply:
revert PR #89217